### ISSUE TITLE
Error spam when adjusting a CollisionShape's position (Godot physics with BVH)

### ISSUE THREAD

**Issue by madmiraal** [godotengine/godot#45180]
**Godot version:**
3.2.4.beta a329de566c109557d6b3b

The bisect points to #45154, but I assume that was hiding the real culprit: #44901.

**OS/device including version:**
Ubuntu 20.10

**Issue description:**
In Godot physics, with the new BVH partitioning, when adjusting a `CollisionShape`'s position, the project freezes with an error spam.
```
ERROR: set_axis_angle: The axis Vector3 must be normalized.
   At: core/math/basis.cpp:980.
ERROR: instance_set_transform: Condition "Math::is_nan(v.x)" is true.
   At: servers/visual/visual_server_scene.cpp:699.
ERROR: instance_set_transform: Condition "Math::is_nan(v.x)" is true.
   At: servers/visual/visual_server_scene.cpp:699.
```

**Steps to reproduce:**
I'm not sure exactly, but it happens when running the attached MRP.

**Minimal reproduction project:**
[BVHBug.zip](https://github.com/godotengine/godot/files/5814005/BVHBug.zip)

**Comment by lawnjelly** [comment]
Ok I'll take a look. :+1: 

In the meantime, you can switch back to old behaviour with `physics->3d->godot_physics->use_bvh` set to off.

**Comment by lawnjelly** [comment]
It seems like the problem is caused by the same CollisionObjectSW being added to the BVH twice.

In the BVH these are being registered as 2 objects with the same userdata pointer, and it calls the pairing callback indicating that they have collided with themselves. This is pretty much correct behaviour from the BVH, though I can easily put in a check and prevent callbacks being called for pairs with the same userdata.

Maybe there is some code in the octree (and bullet?) that prevents pairing between objects with the same userdata. I will investigate and see whether the same double add is occurring with octree.

**Comment by lawnjelly** [comment]
The octree is only adding once. I suspect it may be a simple explanation, that the client code is assuming that 0 ID means the BVH object is unset, whereas actually it is a valid BVH ID. I had to use plus 1 IDs for the rendering, I can easily change this for the physics too. :+1:

**Comment by akien-mga** [comment]
Fixed by #45184.
